tasks/sep/032-op-pectra-defence/NestedSignFromJson.s.sol

```diff
@@ -0,0 +1,138 @@
+    {
+        // Initialize contract references using registry
+        disputeGameFactory = IDisputeGameFactory(proxies.DisputeGameFactory);
+        systemConfig = ISystemConfig(proxies.SystemConfig);
+        proxyAdmin = IProxyAdmin(EIP1967Helper.getAdmin(address(systemConfig)));
+    }
+
+    function setUp() public {
+        addAllowedStorageAccess(address(disputeGameFactory));
+        beforeParams[disputeGameFactory][GameTypes.PERMISSIONED_CANNON] =
+            getGameConstructorParams(IFaultDisputeGame(address(disputeGameFactory.gameImpls(GameTypes.PERMISSIONED_CANNON))));
+
+        beforeParams[disputeGameFactory][GameTypes.CANNON] =
+            getGameConstructorParams(IFaultDisputeGame(address(disputeGameFactory.gameImpls(GameTypes.CANNON))));
+    }
+
+    function _postCheck(Vm.AccountAccess[] memory accesses, Simulation.Payload memory) internal view override {
+        console.log("Running post-deploy assertions");
+
+        // accesses.decodeAndPrint();
+
+        checkStateDiff(accesses);
+
+        // get the before and after game params for the permissioned game
+        IFaultDisputeGame.GameConstructorParams memory beforeParams_ =
+            beforeParams[disputeGameFactory][GameTypes.PERMISSIONED_CANNON];
+        IFaultDisputeGame.GameConstructorParams memory afterParams =
+            getGameConstructorParams(IFaultDisputeGame(address(disputeGameFactory.gameImpls(GameTypes.PERMISSIONED_CANNON))));
+        // Set the before prestate to match the after prestate, since that is the only thing that should have
+        // changed, the two sets of params should now have the same hash.
+        require(
+            Claim.unwrap(beforeParams_.absolutePrestate) != Claim.unwrap(afterParams.absolutePrestate),
+            "Prestate not updated"
+        );
+        require(
+            Claim.unwrap(afterParams.absolutePrestate) == 0x035ac388b5cb22acf52a2063cfde108d09b1888655d21f02f595f9c3ea6cbdcd,
+            "Prestate not updated to expected value"
+        );
+        beforeParams_.absolutePrestate = afterParams.absolutePrestate;
+        require(
+            keccak256(abi.encode(beforeParams_)) == keccak256(abi.encode(afterParams)),
+            "Game params changed unexpectedly"
+        );
+
+        // get the before and after game params for the permissionless game
+        beforeParams_ = beforeParams[disputeGameFactory][GameTypes.CANNON];
+        afterParams = getGameConstructorParams(IFaultDisputeGame(address(disputeGameFactory.gameImpls(GameTypes.CANNON))));
+        // Set the before prestate to match the after prestate, since that is the only thing that should have
+        // changed, the two sets of params should now have the same hash.
+        beforeParams_.absolutePrestate = afterParams.absolutePrestate;
+        require(
+            keccak256(abi.encode(beforeParams_)) == keccak256(abi.encode(afterParams)),
+            "Game params changed unexpectedly"
+        );
```

Comment:
this has less assertions than the permissioned case, it's missing the `"Prestate not updated"` and `"Prestate not updated to expected value"` checks